### ISSUE TITLE
S2259 False Positive with null conditional operator

### ISSUE THREAD

**Issue by nnpcYvIVl** [SonarSource/sonar-dotnet#2361]
The following code flags S2259 on exception after it has been checked by a null-conditional operator. Is this a false positive?

```csharp
using System;
using System.Collections;

namespace S2259_False_Positive
{
    class C
    {
        void M(Exception exception)
        {
            if (exception?.Data is IDictionary data)
            {
                if (exception.InnerException?.Data is IDictionary innerexceptiondata)
                {

                }
            }
        }
    }
}
```

* SonarC# Version: Latest
* Visual Studio Version: 15.9.10

**Comment by andrei-epure-sonarsource** [comment]
Probably #2361 , #2164 and #1837 are all related to how we interpret the null-conditional operator

**Comment by andrei-epure-sonarsource** [comment]
I removed this from 7.15 milestone as there should be a separate milestone dedicated to fixing advanced rules which use Symbolic Execution
